### PR TITLE
remove bootstrap duplicated chain

### PR DESCRIPTION
**Pull request by jackkav**
bootstrap was calling build inconsistently making chained scripts hard to follow.

### CODE REVIEW

**Review by filfreire — APPROVED** [review]

File: packages/insomnia-config/package.json [hunk 1]
```diff
@@ -17,15 +17,14 @@
...
-    "bootstrap": "npm run generate",
     "lint": "eslint . --ext .js,.ts,.tsx --cache",
     "lint:fix": "npm run lint -- --fix",
     "test": "jest",
     "pregenerate": "npm run build",
     "generate": "ts-node ./src/generated/generate.ts",
     "clean": "tsc --build tsconfig.build.json --clean",
     "prebuild": "npm run clean",
-    "build": "tsc --build tsconfig.build.json"
+    "build": "tsc --build tsconfig.build.json && ts-node ./src/generated/generate.ts"
```
Comment (DMarby) [inline packages/insomnia-config/package.json:27]: Just double checking whether the order of this matters? Assuming not

**Review by DMarby — APPROVED** [review]
:shipit:

File: packages/insomnia-config/package.json [hunk 2]
```diff
@@ -17,15 +17,14 @@
   "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
-    "bootstrap": "npm run generate",
```
Comment (dimitropoulos) [inline packages/insomnia-config/package.json:20]: this is a pretty significant change, no? because before it would _not_ build on `bootstrap`, but now it does.

File: packages/insomnia-config/package.json (hunk 1 above)
Comment (dimitropoulos) [reply packages/insomnia-config/package.json:27]: it matters: build first, then generate schema from those built types

Comment (dimitropoulos) [inline packages/insomnia-config/package.json:27]: I think we should be careful about doing this because it's a side effect of the build that we almost definitely do not want to constantly rerun and could also take up lots of unnecessary time in bootstrap

**Review by dimitropoulos — CHANGES_REQUESTED** [review]

File: packages/insomnia/package.json [hunk 3]
```diff
@@ -17,7 +17,6 @@
   "author": "Kong <office@konghq.com>",
   "main": "src/main.min.js",
   "scripts": {
-    "bootstrap": "npm run build:sr",
```
Comment (dimitropoulos) [inline packages/insomnia/package.json:20]: ok, so then when does send request get built?  or did you say today that something changed such that this isn't required anymore?

File: packages/insomnia-config/package.json (hunk 2 above)
Comment (jackkav) [reply packages/insomnia-config/package.json:20]: The npm hooks sure are confusing.
`"pregenerate": "npm run build",`

File: packages/insomnia/package.json (hunk 3 above)
Comment (jackkav) [reply packages/insomnia/package.json:20]: send request is built in CI. This isn't required.

File: packages/insomnia-config/package.json (hunk 1 above)
Comment (jackkav) [reply packages/insomnia-config/package.json:27]: It was already running in bootstrap. I'm just making it more obvious. Since chained scripts are hard to follow.

File: packages/insomnia/package.json (hunk 3 above)
Comment (jackkav) [reply packages/insomnia/package.json:20]: Upon closer inspection I had missed updating the doc, and how app build would work thanks for the catch!

Comment (jackkav) [reply packages/insomnia/package.json:20]: I've made two assertions following this feedback
1. that the send-request abstraction is relevant to inso dx and added it to the dx doc.
2. that all dx/ci flows start with npm run bootstrap therefore ca-cert prep can happen in just one place

File: packages/insomnia-config/package.json (hunk 2 above)
Comment (dimitropoulos) [reply packages/insomnia-config/package.json:20]: sweet.

File: packages/insomnia-config/package.json (hunk 1 above)
Comment (dimitropoulos) [inline packages/insomnia-config/package.json:27]:
```suggestion
    "build": "tsc --build tsconfig.build.json && npm run generate"
```

let's reduce the margin for error here, at least, then.

File: packages/insomnia/package.json [hunk 4]
```diff
@@ -17,12 +17,11 @@
...
     "prebuild": "npm run clean",
-    "build": "npm run generate:ca-certs && npm run build:sr && npm run build:app",
+    "build": "npm run generate:ca-certs",
```
Comment (dimitropoulos) [inline packages/insomnia/package.json:21]: I find it hard to understand how the `build` command in the `insomnia` package doesn't actually build insomnia (anymore), but instead only generates ca-certs??  This is a pretty tough sell on developer experience for me.

Comment (jackkav) [reply packages/insomnia/package.json:21]: open to suggestions

File: packages/insomnia-config/package.json (hunk 1 above)
Comment (jackkav) [reply packages/insomnia-config/package.json:27]: that would create a loop since generate will pregenerate the build step again.

File: packages/insomnia/package.json (hunk 4 above)
Comment (dimitropoulos) [reply packages/insomnia/package.json:21]: I suggest all `build` scripts build the package they belong to (as it did before this PR).  Whatever `build` means for that package can vary (as it does here).

also, it has a strange indirection here now because `prebuild` will clean things that weren't done during the `build`

File: packages/insomnia-config/package.json (hunk 1 above)
Comment (dimitropoulos) [reply packages/insomnia-config/package.json:27]: ok, so then we remove the `pregenerate`?

**Comment by jackkav** [comment]
Since "build" terminology exists for the app, inso, packages and plugins and the meaning is not consistent. Therefore this PR does not entirely solve the problem. A name change may be in order. Abandoning.
